Kconfig.zephyr

```diff
@@ -370,6 +370,12 @@ config COMPILER_WARNINGS_AS_ERRORS
 	help
 	  Turn on "warning as error" toolchain flags
 
+config COMPILER_SAVE_TEMPS
+	bool "Save temporary object files"
+	help
+	  Instruct the compiler to save the temporary object files, which can
+	  be useful for troubleshooting build issues.
```

Comment:
Actually it's more than the object files which are kept, so I suggest to rephrase a bit.

Reply:
Good point, used the wording from https://gcc.gnu.org/onlinedocs/gcc/Developer-Options.html: "Store the usual “temporary” intermediate files permanently"